src/components/todoItem/TodoItem.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { FC, useContext, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { TodoContext } from '../../context/context';
+import { updateTodos } from '../../api/todos';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: FC<Props> = ({ todo }) => {
+  const context = useContext(TodoContext);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [todoStatus, setTodoStatus] = useState<boolean | null>(null);
+
+  const onSelectInputChange = (id: number, completed: boolean) => {
+    setSelectedTodoId(id);
+    setTodoStatus(!completed);
+  };
+
+  useEffect(() => {
+    if (selectedTodoId !== null && todoStatus !== null) {
+      const updateTodo = async () => {
+        try {
+          context?.setTodoLoading(selectedTodoId, true);
+          context?.setErrorMessage('');
+          await updateTodos(selectedTodoId, todoStatus);
+          context?.setTodos(prevTodos =>
+            prevTodos.map(item =>
+              item.id === selectedTodoId
+                ? { ...item, completed: todoStatus }
+                : item,
+            ),
+          );
+        } catch (error) {
+          context?.setErrorMessage('Something went wrong');
+        } finally {
+          context?.setTodoLoading(selectedTodoId, false);
+        }
+      };
+
+      updateTodo();
+    }
+  }, [selectedTodoId, todoStatus]);
```

Comment:
It shouldn't be in `useEffect`, just call the `updateTodo` function when you need it